core/domain/user_domain.py

```diff
@@ -345,3 +406,25 @@ def validate(self):
             raise utils.ValidationError(
                 'Expected can_review_questions to be a boolean value, '
                 'found: %s' % type(self.can_review_questions))
+
+
+class FullyQualifiedUserScoreIdentifier(python_utils.OBJECT):
```

Comment:
This might not be needed -- it depends on what is left once you drop the unneeded plural functions. If the args can be passed separately in the functions without any conceptual issues then let's drop this and pass args separately.